### PR TITLE
[Feature] Use the last modified time of the project file to return document cache

### PR DESCRIPTION
**Pull request by rldhont**
If the time of most recent content modification of document cached is greater than the last modified time of the project
file as returned by the file system or other project storage then the cached is returned else the cached document is removed.

* Funded by Ifremer

### CODE REVIEW

**Review by dmarteau — CHANGES_REQUESTED** [review]

File: wmtsCacheServer/cachefilter.py [hunk 1]
```diff
@@ -82,9 +82,18 @@ def getCachedDocument(self, project: 'QgsProject', request: 'QgsServerRequest',
...
-                with p.open('rb') as f:
-                    self.set_debug_headers(path=p)
-                    return QByteArray(f.read())
+                # time of most recent content modification of document cached
+                dcDateTime = QDateTime.fromSecsSinceEpoch(p.stat().st_mtime)
+                # last modified time of the project file as returned by the file system
+                # or other project storage
+                pDateTime = project.lastModified()
+                # return cached document if it is most recent than the project
+                if dcDateTime > pDateTime:
```
[the remark this answers is not in the document]
Comment (rldhont) [reply wmtsCacheServer/cachefilter.py:91]: @dmarteau in `CacheHelper` or in `DiskCacheFilter` ?

If I implement it in `CacheHelper`, I need to change the API.

[the remark this answers is not in the document]
Comment (rldhont) [reply wmtsCacheServer/cachefilter.py:91]: add `DateTime` in `CacheHelper`

[the remark this answers is not in the document]
Comment (rldhont) [reply wmtsCacheServer/cachefilter.py]: Changes done, ready for review

**Review by dmarteau — COMMENTED** [review]
LGTM

**Review by dmarteau — APPROVED** [review]

**Pull request merged**